### PR TITLE
projinfo: 2 args equivalent to usage of -s and -t

### PR DESCRIPTION
**Pull request by jjimenezshaw**
From
https://lists.osgeo.org/pipermail/proj/2024-May/011388.html

Some times when I am using cs2cs I want to do also a projinfo, to see the
transformation(s) that is may apply.

It is always bothering me that I have to add "-s" and "-t".

Does it make sense that if there are two args without option identifier, it
assumes that the second is "-t"? Am I missing some corner-cases? (if you
like the idea, I am happy to do the PR)
I do not mean to remove "-t" completely, but to make the arg parser more
flexible.

- [x] Tests added
- [x] Added clear title that can be used to generate release notes
- [x] Fully documented, including updating `docs/source/*.rst` for new API

### CODE REVIEW

File: src/apps/projinfo.cpp [hunk 1]
```diff
@@ -1042,8 +1043,7 @@ int main(int argc, char **argv) {
         usage();
     }
 
-    std::string user_string;
-    bool user_string_specified = false;
+    std::vector<std::string> user_strings;
```
Comment (rouault) [inline src/apps/projinfo.cpp]: perhaps ``positional_args`` would be clearer ?

Comment (jjimenezshaw) [reply src/apps/projinfo.cpp]: done

**Review by rouault — APPROVED** [review]
